### ISSUE TITLE
[@typescript-eslint/no-base-to-string] is complaining about boolean values

### ISSUE THREAD

**Issue by juergenzimmermann** [typescript-eslint/typescript-eslint#1655]
Are you opening an issue because the rule you're trying to use is not found?
No

**Repro**

Rule definition in YAML
```YAML
@typescript-eslint/no-base-to-string": error
```

Blamed TypeScript code fragment:
```TS
const result = compareSync(password, user.password); // result is of type boolean
logger.error(`AuthService.checkPassword(): ${result}`);
```

**Expected Result**
No output

**Actual Result**
`152:51  error  'result will evaluate to '[Object object]' when stringified  @typescript-eslint/no-base-to-string`

**Additional Info**

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `2.21.1-alpha.5` |
| `@typescript-eslint/parser`        | `2.21.1-alpha.5` |
| `TypeScript`                       | `3.8.3` |
| `ESLint`                           | `7.0.0-alpha.2` |
| `node`                             | `13.9.0` |
| `npm`                              | `6.14.1` |

**Comment by bradzacher** [comment]
Could you provide more information?
The type defs for everything would be great.

A small, self contained repro would be better.

**Comment by leaumar** [comment]
I can provide a live example: please check [this PR](https://gitlab.com/Marnes/doubleclick-image-downloader/-/merge_requests/35)

The rule is complaining at `Restrictions.ts:59` about a boolean in a template string

![image](https://user-images.githubusercontent.com/3950300/75707114-a16c6a80-5cbe-11ea-9fdc-ac178ae9d131.png)
![image](https://user-images.githubusercontent.com/3950300/75707135-aaf5d280-5cbe-11ea-9f74-c40f8b0bb03c.png)

**Comment by felixfbecker** [comment]
Saw the same - I would expect the `no-base-to-string` to only be used for object types, as `restrict-template-expression` can already take care of the other

**Comment by bradzacher** [comment]
This is a bug.

Self-contained repro is `` `${true}` ``.

The rule grabs the `toString` property's declaration for the given type, and then asserts that the declaration does not come from the base `Object` type.

For some reason, boolean types return `Object` instead of `Boolean`.

For comparison, for both `` `${''}` `` and `` `${1}` `` the owner of the parent type is `String` and `Number`.

It seems like this bug is caused by a bug in typescript itself..

**Comment by bradzacher** [comment]
It is a bug in typescript, the library types are wrong.

`Boolean` has no `toString` defined
https://github.com/microsoft/TypeScript/blob/d209092ccc0ca9fee4fe5148c554484aab95e487/lib/lib.es5.d.ts#L529-L532

Whereas other base types do:
- `Symbol` https://github.com/microsoft/TypeScript/blob/d209092ccc0ca9fee4fe5148c554484aab95e487/lib/lib.es5.d.ts#L99
- `Function` https://github.com/microsoft/TypeScript/blob/d209092ccc0ca9fee4fe5148c554484aab95e487/lib/lib.es5.d.ts#L296
- `String` https://github.com/microsoft/TypeScript/blob/d209092ccc0ca9fee4fe5148c554484aab95e487/lib/lib.es5.d.ts#L396
- `Number` https://github.com/microsoft/TypeScript/blob/d209092ccc0ca9fee4fe5148c554484aab95e487/lib/lib.es5.d.ts#L547
- etc

**Comment by felixfbecker** [comment]
Maybe we should open an issue at the TS repo with a combined list of types with missing `toString()`? `URLSearchParams` also misses it

**Comment by bradzacher** [comment]
feel free if you'd like!

We can't rely on a forward fix for this plugin, as that adds a ~3month turnaround on the fix, and forces people to upgrade their TS version (with breaking changes), so the rule will have to be fixed to workaround regardless.

**Comment by felixfbecker** [comment]
I'd rather have a fix sooner than later too :)

Also noticed with `URL` and `RegExp`

**Comment by bradzacher** [comment]
note that `URL` already has a `toString` defined: https://github.com/microsoft/TypeScript/blob/d209092ccc0ca9fee4fe5148c554484aab95e487/lib/lib.dom.d.ts#L16101
as does `URLSearchParams`: https://github.com/microsoft/TypeScript/blob/d209092ccc0ca9fee4fe5148c554484aab95e487/lib/lib.dom.d.ts#L16152

`RegExp` does not though
